test/e2e/query_test.go

```diff
@@ -1720,6 +1721,43 @@ func rangeQuery(t *testing.T, ctx context.Context, addr string, q func() string,
 	return retExplanation
 }
 
+
+func RemoteWrite(ctx context.Context, timeseries []prompb.TimeSeries, addr string) (*http.Response, error) {
+	// Create write request
+	data, err := proto.Marshal(&prompb.WriteRequest{Timeseries: timeseries})
+	if err != nil {
+		return nil, err
+	}
+
+	// Create HTTP request
+	compressed := snappy.Encode(nil, data)
+	req, err := http.NewRequest("POST", fmt.Sprintf("http://%s/api/v1/receive", addr), bytes.NewReader(compressed))
+	if err != nil {
+		return nil, err
+	}
+
+	req.Header.Add("Content-Encoding", "snappy")
+	req.Header.Set("Content-Type", "application/x-protobuf")
+	req.Header.Set("X-Prometheus-Remote-Write-Version", "0.1.0")
+	// req.Header.Set("X-Scope-OrgID", c.orgID) // Doubt
+
+	// Execute HTTP request
+	res, err := promclient.NewDefaultClient().HTTPClient.Do(req.WithContext(ctx)) // Doubt
+	if err != nil {
+		return nil, err
+	}
+
+	logger := log.NewLogfmtLogger(os.Stdout)
+	logger = log.With(logger, "ts", log.DefaultTimestampUTC)
+	_ = logger.Log(
+		"caller", "remoteWriteFakeMetric",
+		"msg", fmt.Sprintln(res, addr, req, timeseries, err),
+	)
+
+	defer res.Body.Close()
```

Comment:
that seems dubious here if you are returning the response; maybe not returning an response but just an error is better. That way you can check the response status code here and outside just ahs to check the error.